pkg/controllers/vdb/stopdb_reconciler.go

```diff
@@ -120,3 +129,47 @@ func (s *StopDBReconciler) runATCmd(ctx context.Context, initiatorName types.Nam
 		"Successfully stopped the database.  It took %s", time.Since(start).Truncate(time.Second))
 	return nil
 }
+
+// skipStopDB returns true if stop_db is not needed.
+func (s *StopDBReconciler) skipStopDB() bool {
+	if s.PFacts.SandboxName == vapi.MainCluster {
+		return !s.Vdb.IsStatusConditionTrue(vapi.VerticaRestartNeeded)
+	}
+	sb := s.Vdb.GetSandbox(s.PFacts.SandboxName)
+	sbStatus := s.Vdb.GetSandboxStatus(s.PFacts.SandboxName)
+	if sb.Shutdown && sbStatus.Shutdown != sb.Shutdown {
+		return false
+	}
+	return true
+}
+
+// updatesandboxSubclusters updates the shutdown state for the sandbox and its
+// subclusters.
+func (s *StopDBReconciler) updatesandboxSubclusters(ctx context.Context) error {
+	err := vdbstatus.SetSandboxShutdownState(ctx, s.VRec.GetClient(), s.Vdb, s.PFacts.GetSandboxName(), true)
+	if err != nil {
+		return err
+	}
+	// here mark all subclusters in the sandbox for shutdown
+	_, err = vk8s.UpdateVDBWithRetry(ctx, s.VRec, s.Vdb, s.markSandboxSubclustersForShutDown)
+	return err
+}
+
+// markSandboxSubclustersForShutDown marks the subclusters so that the operator
+// does not restart them.
+func (s *StopDBReconciler) markSandboxSubclustersForShutDown() (bool, error) {
```

Comment:
The spec.subclusters.Shutdown should be set no matter whether we stopped the sandbox or not. This should be executed in another reconciler. Here we should update status.subclusters.Shutdown.

Reply:
Here is how I see things. Only 2 entities can update `spec.subclusters[].shutdown`:
- The user, only for stop_sc. They will not be allowed to change that field for a subcluster that belongs to a sandbox with shutdown set to true.
- The operator, after stop_sandbox.

For both cases, there will be a separate reconciler(used for stop_sc) that will update `status.subclusters[].shutdown`. That reconciler will be agnostic to whether we are doing stop_sandbox or stop_sc, and must rely only on `spec.subclusters[].shutdown` and podfacts.
status field must never be updated before the corresponding spec field.